x/stakeibc/keeper/unbonding.go

```diff
@@ -408,40 +369,36 @@ func (k Keeper) UnbondFromHostZone(ctx sdk.Context, hostZone types.HostZone) err
 	}
 
 	// Get the list of relevant records that should unbond
-	_, initialEpochNumberToHostZoneUnbondingMap := k.GetQueuedHostZoneUnbondingRecords(ctx, hostZone.ChainId)
+	epochUnbondingRecordIds, epochNumbersToHostZoneUnbondings := k.GetQueuedHostZoneUnbondingRecords(ctx, hostZone.ChainId)
 
 	// Update the native unbond amount on all relevant records
 	// The native amount is calculated from the stTokens
-	if err := k.RefreshUnbondingNativeTokenAmounts(ctx, initialEpochNumberToHostZoneUnbondingMap); err != nil {
+	epochNumbersToHostZoneUnbondings, err = k.RefreshUnbondingNativeTokenAmounts(ctx, hostZone.ChainId, epochNumbersToHostZoneUnbondings)
+	if err != nil {
 		return err
 	}
 
-	// Fetch the records again with the updated native amounts
-	epochUnbondingRecordIds, epochNumberToHostZoneUnbondingMap := k.GetQueuedHostZoneUnbondingRecords(ctx, hostZone.ChainId)
-
-	// Sum the total number of native tokens that from the records above that are ready to unbond
-	totalUnbondAmount := k.GetTotalUnbondAmount(ctx, epochNumberToHostZoneUnbondingMap)
+	// Sum the total number of native tokens from the records above that are ready to unbond
+	totalNativeUnbondAmount := k.GetTotalUnbondAmount(epochNumbersToHostZoneUnbondings)
 	k.Logger(ctx).Info(utils.LogWithHostZone(hostZone.ChainId,
-		"Total unbonded amount: %v%s", totalUnbondAmount, hostZone.HostDenom))
+		"Total unbonded amount: %v%s", totalNativeUnbondAmount, hostZone.HostDenom))
 
 	// If there's nothing to unbond, return and move on to the next host zone
-	if totalUnbondAmount.IsZero() {
+	if totalNativeUnbondAmount.IsZero() {
 		return nil
 	}
 
-	// Determine the ideal balanced delegation for each validator after the unbonding
-	//   (as if we were to unbond and then rebalance)
-	// This will serve as the starting point for determining how much to unbond each validator
-	// first, get the total delegations _excluding_ validators with slash_query_in_progress
+	// Determine the total eligible unbond amount - excluding delegations to validators with a slash query in progress
 	totalValidDelegationBeforeUnbonding := sdkmath.ZeroInt()
 	for _, validator := range hostZone.Validators {
 		if !validator.SlashQueryInProgress {
 			totalValidDelegationBeforeUnbonding = totalValidDelegationBeforeUnbonding.Add(validator.Delegation)
 		}
 	}
```

Comment:
IDK if related to the PR, but: What are the frequency and duration of a slash query? This might really mess up the rebalancing. Can we sequence it in such a way that once an epoch we do a slash query, then right after we unbond/rebalnce?

Reply:
Also true for https://github.com/Stride-Labs/stride/blob/c9b390dde/x/stakeibc/keeper/host_zone.go#L253-L259

Reply:
So the slash query is a daisy chained query: first one determines if the validator was slashed, and, if so, the second one fires to get the magnitude of the slash. Only the second one is blocking. But the first one can be triggered permissionlessly, and can also be triggered by LSM liquid stakes.

i'm glad you called this out though - it's one of the most (unnecessarily) complex parts of our code base. But I think it's a tough problem cause slash queries can trigger at any time from LSM liquid stakes, and they're UX block when that does occur, so I think they need to take precedence. 

---

I think the best long term approach is actually to use ICQ from ibc-go which would simplify this a ton! We'd be able to just bundle the ICQ with the ICA and then we'd never have to worry about this overlap (here's a [doc](https://docs.google.com/document/d/1gpudfKI8mAfRM2Hix6Lkix8wJN2nzTa-BjHgyW12JMA/edit) for reference). 

Only blocker there though is having host chains [upgrade](https://docs.google.com/spreadsheets/d/18I813HXRCrz3IHq3F4dTAnIe7DaXCxYQWtPoJrCoSdI/edit?gid=0#gid=0)